cmd/fleetctl/user.go

```diff
@@ -172,6 +175,125 @@ func createUserCommand() *cli.Command {
 	}
 }
 
+func createBulkUsersCommand() *cli.Command {
+	return &cli.Command{
+		Name:      "import",
+		Usage:     "Create bulk users",
+		UsageText: `This command will create a set of users in Fleet by imoprting a CSV file.`,
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:     csvFlagName,
+				Usage:    "csv file with all the users (required)",
+				Required: true,
+			},
+			configFlag(),
+			contextFlag(),
+			yamlFlag(),
+			debugFlag(),
+		},
+		Action: func(c *cli.Context) error {
+			client, err := clientFromCLI(c)
+			if err != nil {
+				return err
+			}
+
+			csvFilePath := c.String(csvFlagName)
+
+			csvFile, err := os.Open(csvFilePath)
+			if err != nil {
+				return err
+			}
+			defer csvFile.Close()
+			csvLines, err := csv.NewReader(csvFile).ReadAll()
+
+			if err != nil {
+				return err
+			}
+			for _, record := range csvLines[1:] {
+				name := record[0]
+				email := record[1]
+				password := record[2]
+				sso := c.Bool(record[3])
+				apiOnly := c.Bool(record[4])
+				globalRoleString := c.String(record[5])
+				teamStrings := c.StringSlice(record[6])
+
+				var globalRole *string
+				var teams []fleet.UserTeam
+				if globalRoleString != "" && len(teamStrings) > 0 {
+					return errors.New("Users may not have global_role and teams.")
+				} else if globalRoleString == "" && len(teamStrings) == 0 {
+					globalRole = ptr.String(fleet.RoleObserver)
+				} else if globalRoleString != "" {
+					if !fleet.ValidGlobalRole(globalRoleString) {
+						return fmt.Errorf("'%s' is not a valid team role", globalRoleString)
+					}
+					globalRole = ptr.String(globalRoleString)
+				} else {
+					for _, t := range teamStrings {
+						parts := strings.Split(t, ":")
+						if len(parts) != 2 {
+							return fmt.Errorf("Unable to parse '%s' as team_id:role", t)
+						}
+						teamID, err := strconv.Atoi(parts[0])
+						if err != nil {
+							return fmt.Errorf("Unable to parse team_id: %w", err)
+						}
+						if !fleet.ValidTeamRole(parts[1]) {
+							return fmt.Errorf("'%s' is not a valid team role", parts[1])
+						}
+
+						teams = append(teams, fleet.UserTeam{Team: fleet.Team{ID: uint(teamID)}, Role: parts[1]})
+					}
+				}
+
+				if sso && len(password) > 0 {
+					return errors.New("Password may not be provided for SSO users.")
+				}
+				if !sso && len(password) == 0 {
+					fmt.Print("Enter password for user: ")
+					passBytes, err := terminal.ReadPassword(int(os.Stdin.Fd()))
```

Comment:
If I remember correctly, when you create a user, fleet will ask to reset the password. So maybe that means we don't need this part?

I'm just picturing a user trying to create 20 users typing each password, and then on use 12 they make a mistake and then the command exits with an error, but the previous 11 were created, so now the user has to edit the csv, remove the first 11 and run it again. Not sure if it's ideal UX.

What do you think?

Reply:
I think that should be the UX, but when I removed that section I got an internal authorization error. so I kept that part.  where use can put a dummy password and end-user get's an invitation anyway. but if you could help fix that should be optimal.

Reply:
> If I remember correctly, when you create a user, fleet will ask to reset the password. So maybe that means we don't need this part?

The new user still needs the old password to "reset" the new one.

I don't follow the use case of `len(password) == 0`, this is in case the CSV defined by the `fleetctl` user has empty password column (to disallow passwords stored on a file)?

Reply:
> I'm just picturing a user trying to create 20 users typing each password, and then on use 12 they make a mistake and then the command exits with an error, but the previous 11 were created, so now the user has to edit the csv, remove the first 11 and run it again. Not sure if it's ideal UX.

One option is two for loops. One that process the input and creates a list/slice of users to create. Second for loop actually creates the users from the list/slice.

Reply:
> I think that should be the UX, but when I removed that section I got an internal authorization error. so I kept that part. where use can put a dummy password and end-user get's an invitation anyway. but if you could help fix that should be optimal.

How about when you create a user, you generate a password randomly and print something like this to stdout:

```
User: someUsername               Generated password: jJ*4u2AQO
User: someOtherUsername     Generated password: pE92$@j_W
```

So that the admin creating the users can email each user their unique (one time) password afterwards.